src/Transaction.php

```diff
@@ -5,69 +5,54 @@
 class Transaction {
 
     /**
-     * @var string The domain of the transaction.
+     * @var string The required keys for the transaction.
      */
-    private $domain;
+    private $requiredKeys = [
+        'domain', 
+        'url', 
+        'name', 
+        'articleEID', 
+        'magazineKey', 
+        'checkoutToken'
+    ];
 
     /**
-     * @var string The URL of the transaction.
+     * @var string The payload of the transaction.
      */
-    private $url;
+    private $payload;
 
     /**
-     * @var string The name associated with the transaction.
-     */
-    private $name;
-
-    /**
-     * @var string The external article ID.
+     * NewsLadderTransaction constructor.
+     * 
+     * @param string $payload ['domain', 'url', 'name', 'articleEID', 'magazineKey', 'checkoutToken']
      */
-    private $external_article_id;
+    public function __construct(array $payload) {
+        if (empty($payload)) {
+            throw new \InvalidArgumentException("Payload is empty");
+        }
 
-    /**
-     * @var string The magazine key.
-     */
-    private $magazine_key;
+        if (count($this->requiredKeys) < count($payload)) {
```

Comment:
Is this necessary? We could just ignore additional keys as those aren't used anyways...

Reply:
But it is in the payload to the Service.

Reply:
Yeah, then we should filter the payload by the required keys:

```php
$this->payload = array_intersect_key($payload, array_flip($this->requiredKeys));
```